pkg/cli/images_all_test.go

```diff
@@ -0,0 +1,246 @@
+				All:    true,
+				Quiet:  false,
+				Output: "",
+			},
+			prepare: func(f *mocks.MockClient) {
+				f.EXPECT().ImageList(gomock.Any()).Return(
+					[]apiv1.Image{
+						image1Tagged,
+						image2Untagged,
+					}, nil)
+				allImagesHaveContaines(f)
+			},
+			args: args{
+				args: []string{"-c"},
+			},
+			wantErr: false,
+			wantOut: "REPOSITORY          TAG       IMAGE-ID          CONTAINER                      DIGEST\nproject1::testtag   latest    image-1-9102390   test-image-running-container   test-image-running-container\n",
+		},
+		{
+			name: "acorn images -A -c -a, one tagged", fields: fields{
+				All:    true,
+				Quiet:  false,
+				Output: "",
+			},
+			prepare: func(f *mocks.MockClient) {
+				f.EXPECT().ImageList(gomock.Any()).Return(
+					[]apiv1.Image{
+						image1Tagged,
+						image2Untagged,
+					}, nil)
+				allImagesHaveContaines(f)
+			},
+			args: args{
+				args: []string{"-c", "-a"},
+			},
+			wantErr: false,
+			wantOut: "REPOSITORY          TAG       IMAGE-ID          CONTAINER                      DIGEST\nproject1::testtag   latest    image-1-9102390   test-image-running-container   test-image-running-container\n<none>              <none>    image-2-2198031   test-image-running-container   test-image-running-container\n",
+		},
+		{
+			name: "acorn images -A -c -a, no tags", fields: fields{
+				All:    true,
+				Quiet:  false,
+				Output: "",
+			},
+			prepare: func(f *mocks.MockClient) {
+				f.EXPECT().ImageList(gomock.Any()).Return(
+					[]apiv1.Image{
+						image1Untagged,
+					}, nil)
+				allImagesHaveContaines(f)
+			},
+			args: args{
+				args: []string{"-c", "-a"},
+			},
+			wantErr: false,
+			wantOut: "REPOSITORY   TAG       IMAGE-ID          CONTAINER                      DIGEST\n<none>       <none>    image-1-9102390   test-image-running-container   test-image-running-container\n",
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			//Mocked client for cli's client calls.
+			mClient := mocks.NewMockClient(ctrl)
+			if tt.prepare != nil {
+				tt.prepare(mClient)
+			}
+
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+			// Mock client factory just returns the gomock client.
+			command := CommandContext{
+				ClientFactory: &testdata.MockClientFactoryManual{
+					Client:         mClient,
+					ProjectOptions: project.Options{AllProjects: tt.fields.All},
+				},
+				StdOut: w,
+				StdErr: w,
+				StdIn:  strings.NewReader(""),
+			}
+			tt.args.cmd = NewImage(command)
+			tt.args.cmd.SetArgs(tt.args.args)
+
+			err := tt.args.cmd.Execute()
+
+			if err != nil && !tt.wantErr {
+				assert.Failf(t, "got err when err not expected", "got err: %s", err.Error())
+			} else if err != nil && tt.wantErr {
+				assert.Equal(t, tt.wantOut, err.Error())
+			} else {
+				err := w.Close()
+				if err != nil {
+					t.Fatal(err)
+				}
+				out, _ := io.ReadAll(r)
+				assert.Equal(t, tt.wantOut, string(out))
+			}
```

Comment:
Doesn't look like this is checking for the case where we wanted an error but did not get one.